### PR TITLE
[SMALLFIX] Update Netty minor version

### PR DESCRIPTION
**Pull request by calvinjia**
Requires the changes to remove usages of chmv8. Version >= 4.0.50 is required to support shading without needing to specify system properties.

### CODE REVIEW

**Comment by apc999** [comment]
@calvinjia , when I am compiling the module `core/client/runtime`, I see two versions of netty pulled in:

```
[INFO] --- maven-shade-plugin:2.4.3:shade (uber-jar) @ alluxio-core-client-runtime ---
...
[INFO] Including io.netty:netty:jar:3.7.0.Final in the shaded jar.
...
[INFO] Including io.netty:netty-all:jar:4.0.29.Final in the shaded jar.
```
where `io.netty:netty:jar:3.7.0.Final:compile` is brought by zookeeper.
do we need to fix this too?

**Comment by calvinjia** [comment]
Jenkins, test this please.

File: core/client/runtime/pom.xml [hunk 1]
```diff
@@ -90,6 +90,11 @@
             <configuration>
               <finalName>${project.artifactId}-${project.version}-jar-with-dependencies</finalName>
               <relocations>
+                <relocation>
+                  <pattern>META-INF/native/libnetty_transport_native_epoll_x86_64.so</pattern>
```
Comment (apc999) [inline core/client/runtime/pom.xml]: can you put a javadoc here explaining why we need this relocation and possibly link to some sources if you think that is helpful?

File: core/common/src/test/java/alluxio/EmbeddedChannels.java [hunk 2]
```diff
@@ -50,8 +50,6 @@ public EmbeddedEmptyCtorChannel() {
       ChannelPipeline p = super.pipeline();
       // Remove the dummy handler.
       p.removeFirst();
-      // Remove the LastInboundHandler in super;
-      p.removeLast();
```
Comment (apc999) [inline core/common/src/test/java/alluxio/EmbeddedChannels.java]: why is this no longer needed?

File: pom.xml [hunk 3]
```diff
@@ -316,7 +316,7 @@
       <dependency>
         <groupId>io.netty</groupId>
         <artifactId>netty-all</artifactId>
-        <version>4.0.29.Final</version>
+        <version>4.0.53.Final</version>
```
Comment (apc999) [inline pom.xml]: can you javadoc here we need this version or later to ensure shading works on netty?

File: core/common/src/test/java/alluxio/EmbeddedChannels.java (hunk 2 above)
Comment (calvinjia) [reply core/common/src/test/java/alluxio/EmbeddedChannels.java]: We were using a hack to get around some limitations before.

https://alluxio.atlassian.net/browse/ALLUXIO-3066

File: pom.xml (hunk 3 above)
Comment (calvinjia) [reply pom.xml]: Updated, thanks.

File: core/client/runtime/pom.xml (hunk 1 above)
Comment (calvinjia) [reply core/client/runtime/pom.xml]: Updated, thanks.

**Comment by calvinjia** [comment]
@apc999 We shouldn't have any issues between netty 3 and netty 4 because the package names are different (org.jboss vs io.netty)

**Pull request merged**